packages/react-native/React/Base/RCTRootView.m

```diff
@@ -432,6 +433,46 @@ - (void)viewDidChangeEffectiveAppearance
 #endif // macOS]
 
 
+#pragma mark - Key window blur/focus
+
+#if TARGET_OS_OSX // [macOS
+- (void)viewDidMoveToWindow {
+  [super viewDidMoveToWindow];
+
+  NSWindow *window = [self window];
+  if (window == nil) {
+    return;
+  }
+
+  [[NSNotificationCenter defaultCenter] addObserver:self
```

Comment:
`[non-blocking]` it would be a pretty advanced case, but if the view was re-parented to another window we would likely want to unregister the previous notifications?

Okay to not resolve this tricky situation but possibly worth a comment if the re-parenting consideration is valid

Reply:
I was going based on a snippet from [Apple's documentation](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1415360-addobserver?language=objc#discussion):

> If your app targets iOS 9.0 and later or macOS 10.11 and later, you do not need to unregister an observer that you created with this function. If you forget or are unable to remove an observer, the system cleans up the next time it would have posted to it.

That being said, it's not immediately clear how the system cleans up these stale observers, so if this particular advanced case can happen, I suppose it's not unreasonable to keep track of an extra bit of state. Still, this largely depends on the internals of `NSNotificationCenter`. I'll ask around.

Reply:
I think that tracks for me with regards to RCTRootViews that are destroyed before the NSWindow (e.g. in-canvas) or NSWindow destroyed without the RCTRootView actually being destroyed (no known scenario this would matter), but if an RCTRootView was directly re-parented across two NSWindows with none of them being destroyed, would it be registered to notifications from both NSWindows?

It's a pretty contrived case regardless; definitely not looking for a change to this PR, just increasing my understanding of relevant concerns :)

Reply:
Yeah, this sounds like something we can follow up on later. For now, let's ship it.